compaction/compaction_manager.cc

```diff
@@ -1127,12 +1127,21 @@ future<> compaction_manager::drain() {
 }
 
 future<> compaction_manager::stop() {
-    do_stop();
-    if (auto cm = std::exchange(_task_manager_module, nullptr)) {
-        co_await cm->stop();
-    }
-    if (_state != state::none) {
-        co_return co_await std::move(*_stop_future);
+    try {
+        do_stop();
+        if (auto cm = std::exchange(_task_manager_module, nullptr)) {
+            auto f = cm->stop();
```

Comment:
compaction_manager::stop() (and all stop() methods) should be responsible for their own error handling. Just like close() methods do. If we need to collect errors, there should be a flush() method called ahead.